gss-serv-krb5.c

```diff
@@ -85,26 +112,62 @@ static int
 ssh_gssapi_krb5_userok(ssh_gssapi_client *client, char *name)
 {
 	krb5_principal princ;
-	int retval;
+	int retval, matched;
 	const char *errmsg;
 
 	if (ssh_gssapi_krb5_init() == 0)
 		return 0;
 
-	if ((retval = krb5_parse_name(krb_context, client->exportedname.value,
-	    &princ))) {
+	retval = krb5_parse_name(krb_context, client->exportedname.value, &princ);
+	if (retval) {
 		errmsg = krb5_get_error_message(krb_context, retval);
 		logit("krb5_parse_name(): %.100s", errmsg);
 		krb5_free_error_message(krb_context, errmsg);
 		return 0;
 	}
-	if (krb5_kuserok(krb_context, princ, name)) {
-		retval = 1;
-		logit("Authorized to %s, krb5 principal %s (krb5_kuserok)",
-		    name, (char *)client->displayname.value);
-	} else
+
+	retval = krb5_kuserok(krb_context, princ, name) == TRUE;
```

Comment:
the logic is a little hard to follow here.

Maybe it would be clearer (and safer) to have a dedicated `success` variable that is explicitly set only when the function has succeeded. I.e. after the `options.gss_indicators == NULL` test below and after `ssh_gssapi_check_indicators()`  returns a positive match.